theories/datatypes/FSet.ec

```diff
@@ -586,6 +586,14 @@ proof.
   rewrite eq_sym eqEcard z_cA fcards0 //=; apply/sub0set.
 qed.
 
+lemma fcard_eq1 (A : 'a fset) : (card A = 1) <=> (exists a, A = fset1 a).
+proof.
+  split=> [o_cA|]; last by move=> [] a ->; apply/fcard1.
+  exists (pick A); rewrite eq_sym eqEcard o_cA fcard1 //=.
+  move=> a /in_fset1 ->; apply: mem_pick.
+  by rewrite -fcard_eq0 o_cA.
+qed.
```

Comment:
I think adding the missing `sub1set` lemma yields a shorter and more idiomatic proof
```suggestion
lemma sub1set x (A : 'a fset) : fset1 x \subset A <=> x \in A.
proof.
rewrite subsetP; split => [|x_A y /in_fset1 -> //].
by apply; rewrite in_fset1.
qed.

lemma fcard_eq1 (A : 'a fset) : (card A = 1) <=> (exists a, A = fset1 a).
proof.
split=> [o_cA|[a ->]]; last apply/fcard1.
exists (pick A); rewrite eq_sym eqEcard o_cA fcard1 /= sub1set mem_pick.
by apply: contraL o_cA => ->; rewrite fcards0.
qed.
```

Reply:
We should merge. If we want to realign, we can do it in a later PR.